Content.Client/Xenoarchaeology/Ui/NodeScannerDisplay.xaml.cs

```diff
@@ -0,0 +1,71 @@
+using Content.Client.UserInterface.Controls;
+using Content.Shared.Xenoarchaeology.Equipment.Components;
+using Robust.Client.AutoGenerated;
+using Robust.Client.UserInterface.Controls;
+using Robust.Client.UserInterface.XAML;
+
+namespace Content.Client.Xenoarchaeology.Ui;
+
+[GenerateTypedNameReferences]
+public sealed partial class NodeScannerDisplay : FancyWindow
+{
+    [Dependency] private readonly IEntityManager _ent = default!;
+
+    public NodeScannerDisplay(EntityUid scannerEntityUid)
+    {
+        RobustXamlLoader.Load(this);
+
+        IoCManager.InjectDependencies(this);
+
+        var scannerComponent = _ent.GetComponent<NodeScannerComponent>(scannerEntityUid);
+
+        Update((scannerEntityUid, scannerComponent));
+    }
+
+    public void Update(Entity<NodeScannerComponent> ent)
+    {
+        var triggeredNodesSnapshot = ent.Comp.TriggeredNodesSnapshot;
+        if (triggeredNodesSnapshot.Count == 0)
+        {
+            NoActiveNodeDataLabel.Visible = true;
+            ActiveNodesList.Visible = false;
+            ActiveNodesList.Children.Clear();
+        }
+        else
+        {
+            NoActiveNodeDataLabel.Visible = false;
+            ActiveNodesList.Visible = true;
+
+            ActiveNodesList.Children.Clear();
+
+            foreach (var nodeId in triggeredNodesSnapshot)
+            {
+                var nodeLabel = new Button
+                {
+                    Text = nodeId,
+                    Margin = new Thickness(15, 5, 0, 0),
+                    MaxHeight = 40,
+                    Disabled = true
+                };
+                ActiveNodesList.Children.Add(nodeLabel);
+            }
+        }
+
+        ArtifactStateLabel.Text = GetState(ent);
```

Comment:
very nice idea. 
considering options of using return here with else - i'd prefer to changing it so that first if blockis the one that contains most logic (where there is stuff to actually render!) and then if-else looks more intuitive, like the first branch is the main, the second is for boring cases. Adding 'return' just to avoid 'else' is not automatically good option.